concourse/steps/release.py

```diff
@@ -209,93 +200,169 @@ def __init__(
         else:
             self.release_commit_callback = None
 
-        if next_version_callback:
-            self.next_version_callback = os.path.join(
-                repo_dir_absolute,
-                next_version_callback,
-            )
-        else:
-            self.next_version_callback = None
+    def _release_commit_message(self, version: str):
+        return f'Release {version}'
 
     def name(self):
-        return "Create Release Commits"
+        return 'Create Release Commit'
 
     def validate(self):
         existing_dir(self.repo_dir)
         semver.parse(self.release_version)
-
         if(self.release_commit_callback):
             existing_file(self.release_commit_callback)
-        if self.next_version_callback:
-            existing_file(self.next_version_callback)
 
         existing_file(self.repository_version_file_path)
 
-        # perform version ops once to validate args
-        self._calculate_next_cycle_dev_version(
-            release_version=self.release_version,
-            version_operation=self.version_operation,
-            prerelease_suffix=self.prerelease_suffix,
-        )
-
-    def _release_commit_message(self, version: str):
-        return f'Release {version}'
-
-    def _next_dev_cycle_commit_message(self, version: str):
-        return f'Prepare next dev cycle {version}'
-
     def apply(self):
         # clean repository if required
         worktree_dirty = bool(self.git_helper._changed_file_paths())
         if worktree_dirty:
             self.git_helper.repo.head.reset(working_tree=True)
 
-        # first, prepare release commit
+        # prepare release commit
         with open(self.repository_version_file_path, 'w') as f:
             f.write(self.release_version)
 
         # call optional release commit callback
         if self.release_commit_callback:
-            self._invoke_callback(
+            _invoke_callback(
```

Comment:
why did you convert this into a module function?

Reply:
Yes it's because of code duplication. `_invoke_callback` is a generic function, which is called once for a Release Commit Callback and once for Next Dev Cycle Commit Callback.

Reply:
ok, understood